osu.Game/Overlays/News/Sidebar/YearsPanel.cs

```diff
@@ -0,0 +1,176 @@
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using System.Collections.Generic;
+using osu.Game.Graphics;
+using osu.Framework.Bindables;
+using osu.Game.Online.API.Requests.Responses;
+using System;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.News.Sidebar
+{
+    public class YearsPanel : CompositeDrawable
+    {
+        private readonly Bindable<APINewsSidebar> metadata = new Bindable<APINewsSidebar>();
+
+        private Container gridPlaceholder;
+
+        [BackgroundDependencyLoader]
+        private void load(OverlayColourProvider colourProvider, Bindable<APINewsSidebar> metadata)
+        {
+            this.metadata.BindTo(metadata);
+
+            AutoSizeAxes = Axes.Y;
+            RelativeSizeAxes = Axes.X;
+            Masking = true;
+            CornerRadius = 6;
+            InternalChildren = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colourProvider.Background3
+                },
+                gridPlaceholder = new Container
+                {
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Padding = new MarginPadding(5)
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            metadata.BindValueChanged(m =>
+            {
+                if (m.NewValue == null)
+                {
+                    Hide();
+                    return;
+                }
+
+                gridPlaceholder.Child = new YearsGridContainer(m.NewValue.Years, m.NewValue.CurrentYear);
+                Show();
+            }, true);
+        }
+
+        private class YearButton : OsuHoverContainer
+        {
+            protected override IEnumerable<Drawable> EffectTargets => new[] { text };
+
+            private readonly OsuSpriteText text;
+            private readonly bool isCurrent;
+
+            public YearButton(int year, bool isCurrent)
+            {
+                this.isCurrent = isCurrent;
+
+                RelativeSizeAxes = Axes.X;
+                Height = 15;
+                Child = text = new OsuSpriteText
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Font = OsuFont.GetFont(size: 12, weight: isCurrent ? FontWeight.SemiBold : FontWeight.Medium),
+                    Text = year.ToString()
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OverlayColourProvider colourProvider)
+            {
+                IdleColour = isCurrent ? Color4.White : colourProvider.Light2;
+                HoverColour = isCurrent ? Color4.White : colourProvider.Light1;
+                Action = () => { }; // Avoid button being disabled since there's no proper action assigned.
```

Comment:
As mentioned in comment, we probably want this to do more than nothing.

Reply:
Relies on integration since we will be calling a method in a NewsOverlay which will trigger a new request, which will reconstruct the whole side bar.

Reply:
What I'm saying is that in the test scene this should be hooked up locally. So that it doesn't do nothing.

Reply:
Should be resolved in https://github.com/ppy/osu/pull/12736/commits/586c5c7365b3d0d0291447fb57b316b1e01b098e